File: chain-sdk/clojure/src/on_chain_api/specs/burn_v3_dto.clj
(ns on-chain-api.specs.burn-v3-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def burn-v3-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :id) string?
   (ds/opt :transaction) string?
   (ds/opt :pool) string?
   (ds/opt :token_0) string?
   (ds/opt :token_1) string?
   (ds/opt :timestamp) string?
   (ds/opt :owner) string?
   (ds/opt :origin) string?
   (ds/opt :amount) string?
   (ds/opt :amount_0) string?
   (ds/opt :amount_1) string?
   (ds/opt :amount_usd) string?
   (ds/opt :tick_lower) string?
   (ds/opt :tick_upper) string?
   (ds/opt :log_index) string?
   (ds/opt :vid) int?
   })

(def burn-v3-dto-spec
  (ds/spec
    {:name ::burn-v3-dto
     :spec burn-v3-dto-data}))

File: chain-sdk/javascript-closure-angular/API/Client/BurnV3DTO.js
goog.provide('API.Client.BurnV3DTO');

/**
 * Burn entities are created for every emitted Burn event on the Uniswap core contracts. The Burn entity stores key data about the event like token amounts, who burned, who received tokens, and more. This entity can be used to track liquidity removals on pairs.
 * @record
 */
API.Client.BurnV3DTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.BurnV3DTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.BurnV3DTO.prototype.recvTime;

/**
 * Number of block in which entity was recorded.
 * @type {!number}
 * @export
 */
API.Client.BurnV3DTO.prototype.blockNumber;

/**
 * Transaction hash + '#' + index in mints Transaction array.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.id;

/**
 * Transaction burn was included in.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.transaction;

/**
 * Pool position is within.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.pool;

/**
 * Reference to token0 as stored in pool contract.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.token0;

/**
 * Reference to token1 as stored in pool contract.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.token1;

/**
 * Timestamp.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.timestamp;

/**
 * Owner of position where liquidity was burned.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.owner;

/**
 * Transaction origin: the EOA (Externally Owned Account) that initiated the transaction.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.origin;

/**
 * Amount of liquidity burned.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.amount;

/**
 * Amount of token 0 burned.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.amount0;

/**
 * Amount of token 1 burned.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.amount1;

/**
 * Derived amount based on available prices of tokens.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.amountUsd;

/**
 * Lower tick of position.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.tickLower;

/**
 * Upper tick of position.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.tickUpper;

/**
 * Position within the transactions.
 * @type {!string}
 * @export
 */
API.Client.BurnV3DTO.prototype.logIndex;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.BurnV3DTO.prototype.vid;

